app/client/src/workers/formEval.ts

```diff
@@ -0,0 +1,133 @@
+}
+
+// Object to hold the initial eval object
+let finalEvalObj: { [x: string]: { visible: boolean; conditionals: any } };
+
+// Recursive function to generate the evaluation state for form config
+const generateInitialEvalState = (formConfig: any) => {
+  const visible = false;
+
+  // Any element is only added to the eval state if they have a conditional statement present, if not they are allowed to be rendered
+  if (formConfig.hasOwnProperty("conditionals")) {
+    let key = "unknowns";
+
+    // A unique key is used to refer the object in the eval state, can be propertyName, configProperty or identifier
+    if (formConfig.hasOwnProperty("propertyName")) {
+      key = formConfig.propertyName;
+    } else if (formConfig.hasOwnProperty("configProperty")) {
+      key = formConfig.configProperty;
+    } else if (formConfig.hasOwnProperty("identifier")) {
+      key = formConfig.identifier;
+    }
+
+    // Conditionals are stored in the eval state itself for quick access
+    finalEvalObj[key] = {
+      visible,
+      conditionals: formConfig.conditionals,
+    };
+  }
+
+  if (formConfig.children)
+    formConfig.children.forEach((config: any) =>
+      generateInitialEvalState(config),
+    );
+};
+
+// Generator function to run the eval for the whole form when data changes
+function evaluate(
+  actionConfiguration: any,
+  currentEvalState: FormEvaluationState,
+) {
+  Object.keys(currentEvalState).forEach((key: string) => {
+    try {
+      if (currentEvalState[key].hasOwnProperty("conditionals")) {
+        const conditionBlock = currentEvalState[key].conditionals;
+        Object.keys(conditionBlock).forEach((conditionType) => {
+          const output = eval(conditionBlock[conditionType]);
+          if (conditionType === ConditionType.HIDE) {
+            currentEvalState[key].visible = !output;
+          } else if (conditionType === ConditionType.SHOW) {
+            currentEvalState[key].visible = output;
+          }
+        });
+      }
+    } catch (e) {
+      log.error(e);
+      Sentry.captureException(e);
+    }
+  });
+  return currentEvalState;
+}
+
+// Fetches current evaluation and runs a new one based on the new data
+function getFormEvaluation(
+  formId: string,
+  actionConfiguration: any,
+  currentEvalState: any,
+): any {
+  // Only change the form evaluation state if the form ID is same or the evaluation state is present
+  if (!!currentEvalState && currentEvalState.hasOwnProperty(formId)) {
+    currentEvalState[formId] = evaluate(
+      actionConfiguration,
+      currentEvalState[formId],
+    );
+
+    // yield put({
+    //   type: ReduxActionTypes.SET_FORM_EVALUATION,
+    //   payload: currentEvalState,
+    // });
+  }
+
+  return currentEvalState;
+}
+
+// Filter function to assign a function to the Action dispatched
+export function setFormEvaluationSaga(
+  type: string,
+  payload: any,
+  currentEvalState: any,
+) {
+  if (type === ReduxActionTypes.INIT_FORM_EVALUATION) {
```

Comment:
```suggestion
  if (type === EVAL_WORKER_ACTIONS.INIT_FORM_EVALUATION) {
```

Why is this check needed?

Reply:
This is because there are two stages to eval. At first, the eval structure is extracted from the JSON forms, then the evals are run in the next step. This check is for that purpose only.